### DISCUSSION TITLE
merge inspect(engine) and meta.reflect(conn) into the same object

### DISCUSSION THREAD

**Discussion by gertcuykens** [sqlalchemy/sqlalchemy#8454] [Ideas]
Can they both result in the same object to simplify things?

```meta = inspect(engine)```
```meta.reflect(conn)```

So no separate inspector, just metadata in 2.0

**Comment by zzzeek** [comment]
MetaData is a stateful collection of Table objects, and Inspector is an object that links to an engine to provide raw DDL catalog queries, I don't see how these functions overlap.     Inspector references an Engine or Connection, while MetaData does not; we are just finishing a multiyear deprecation cycle to remove MetaData.bind.  So your proposal to remove Inspector would also require reversing that entire decision.

An API I could see happening would be augmenting metadata.reflect(conn) with inspector.reflect_into(metadata) and reflect_into(Table), so that we remove the entire concept of engines/connections ever being passed to MetaData/Table, but I dont think it's necessary to do that.

But overall Insepctor is an object that deals with connections and runs queries, MetaData is an object that deals with Python structures that represent tables.  These are two separate concerns.

**Reply by CaselIT** [reply]
Also engine doesn't have any link to a metadata object, so I don't know how `meta = inspect(engine)` could work

**Reply by gertcuykens** [reply]
ok thanks, the only reason I ask is to prevent bad 2.0 tutorials in the wild, as in where they do the same thing using the inspector while they should be using metadata to create a table or something.

**Reply by CaselIT** [reply]
Do you have an example?

The inspector can [reflect a single table](https://docs.sqlalchemy.org/en/20/core/reflection.html?highlight=inspector#sqlalchemy.engine.reflection.Inspector.reflect_table), but the rest of the api is a lot more low level: https://docs.sqlalchemy.org/en/20/core/reflection.html?highlight=inspector#sqlalchemy.engine.reflection.Inspector

**Reply by gertcuykens** [reply]
no example found, I guess it's fine. I assume it can't be made anymore consistent then this
